File: go/0009.go
package leetcode

func isPalindrome(x int) bool {

	if x < 0 {
		return false
	}

	y := make([]int, 0)

	for x > 0 {
		y = append(y, x%10)
		x = int(x / 10)
	}

	for i := 0; i < len(y)/2; i++ {
		if y[i] != y[len(y)-1-i] {
			return false
		}
	}

	return true
}

File: java/0009.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0))
            return false;

        int palindrome = 0, xTemp = x;
        while (xTemp > 0) {
            palindrome = palindrome * 10 + xTemp % 10;
            xTemp /= 10;
        }
        return x == palindrome;
    }
}